crates/primitives/src/chains/mod.rs

```diff
@@ -0,0 +1,66 @@
+use crate::{ForkId, Hardfork, H256};
+
+pub use specs::{ChainSpec, Essentials, NetworkUpgrades};
+
+mod custom;
+mod goerli;
+mod mainnet;
+mod sepolia;
+mod specs;
+
+pub use goerli::GoerliSpec;
+pub use mainnet::MainnetSpec;
+pub use sepolia::SepoliaSpec;
+
+use self::custom::{CustomChainSpec, CustomChainSpecBuilder};
+
+#[derive(Debug, Clone, Copy)]
+pub enum ChainSpecUnified {
```

Comment:
Indeed, I added this enum for a bad reason: I can't use `Box<dyn ChainSpec>` because `clap::ArgMatches` has to have `Clone`, so `ChainSpec` must be `Clone` and trait objects can't.

It seems to go in favor of your argument to make `ChainSpec` a struct in a other comment...